### DISCUSSION TITLE
Could the symbol picker be used to navigate documents using Markdown headings?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#2310] [Ideas] [answered]
It would be great to be able to navigate markdown using the headings:
```
h1 heading1
   h2 heading2
       h3 heading3
   h2 heading4
   h2 heading5
```
Is that in the scope of the symbol picker?

This feature would be a game changer. Just having an overview of the document headings alone would be really useful.

**Comment by the-mikedavis** [comment, marked as answer]
Ideally this is something an LSP for markdown would provide (I need to swing back around to https://github.com/helix-editor/helix/pull/2182 😅)

This could alternatively be accomplished with tree-sitter tags (https://github.com/helix-editor/helix/issues/1252 is about stack graphs but the concepts are similar). ~I think that's what [Zas editor](https://www.zeditor.app/) is doing with the "Filter Symbols" picker (scroll down to the "file structure" part of zas' website).~ For programming languages symbols are usually functions / methods / classes but for markup you could tag headers and paragraphs and the like.

**Comment by danielgomez3** [comment]
I'm getting this feature to work using the symbol picker with marksman lsp in Helix!
